charts/incubator/plextraktsync/values.yaml

```diff
@@ -1,28 +1,90 @@
-env: {}
 image:
   pullPolicy: IfNotPresent
   repository: tccr.io/truecharts/plextraktsync
-  tag: latest@sha256:dbbbc221c97fd7ff1b142a167709f66962c4f370cb5f40839670767cd541c41f
-persistence: {}
+  tag: latest@sha256:55d0b5927a1bc11dd83a94cd56375a201de3dc03f099345b71404ca20757d373
+
+plexTraktSync:
+  PLEX_BASEURL: "http://plex.ix-plex.svc.cluster.local:32400"
+  PLEX_FALLBACKURL: "http://plex.ix-plex.svc.cluster.local:32400"
+  PLEX_LOCALURL: "http://localhost:32400"
+  PLEX_TOKEN: "MY_PLEX_TOKEN"
+  PLEX_OWNER_TOKEN: ""
+  PLEX_USERNAME: "MY_PLEX_USERNAME"
+  TRAKT_USERNAME: "MY_TRAKT_USERNAME"
+
 podSecurityContext:
   runAsGroup: 0
   runAsUser: 0
+
+securityContext:
+  readOnlyRootFilesystem: false
+  runAsNonRoot: false
+
 probes:
   liveness:
     enabled: false
   readiness:
     enabled: false
   startup:
     enabled: false
-securityContext:
-  readOnlyRootFilesystem: false
-  runAsNonRoot: false
+
 service:
   main:
     enabled: false
     ports:
       main:
         enabled: false
 
+persistence:
+  config:
+    enabled: true
+    mountPath: "/app/config"
+
+initContainers:
+  create-config:
+    name: create-config
+    image: "{{ .Values.alpineImage.repository }}:{{ .Values.alpineImage.tag }}"
+    volumeMounts:
+      - name: config
+        mountPath: "/app/config"
+    env:
+      - name: PLEX_BASEURL
+        value: "{{ .Values.plexTraktSync.PLEX_BASEURL }}"
+      - name: PLEX_FALLBACKURL
+        value: "{{ .Values.plexTraktSync.PLEX_FALLBACKURL }}"
+      - name: PLEX_LOCALURL
+        value: "{{ .Values.plexTraktSync.PLEX_LOCALURL }}"
+      - name: PLEX_TOKEN
+        value: "{{ .Values.plexTraktSync.PLEX_TOKEN }}"
+      - name: PLEX_OWNER_TOKEN
+        value: "{{ .Values.plexTraktSync.PLEX_OWNER_TOKEN }}"
+      - name: PLEX_USERNAME
+        value: "{{ .Values.plexTraktSync.PLEX_USERNAME }}"
+      - name: TRAKT_USERNAME
+        value: "{{ .Values.plexTraktSync.TRAKT_USERNAME }}"
+    command: ["/bin/sh", "-c"]
+    args:
+      - >
+        echo "Creating config files...";
+        export env=/app/config/.env;
+        export pytrakt=/app/config/.pytrakt.json;
+        if [ ! -f $env ]; then
+          echo "# This is .env file for PlexTraktSync" > $env;
+          echo "PLEX_BASEURL=$PLEX_BASEURL" >> $env;
+          echo "PLEX_FALLBACKURL=$PLEX_FALLBACKURL" >> $env;
+          echo "PLEX_LOCALURL=$PLEX_LOCALURL" >> $env;
+          echo "PLEX_TOKEN=$PLEX_TOKEN" >> $env;
+          echo "PLEX_OWNER_TOKEN=$PLEX_OWNER_TOKEN" >> $env;
+          echo "PLEX_USERNAME=$PLEX_USERNAME" >> $env;
+          echo "TRAKT_USERNAME=$TRAKT_USERNAME" >> $env;
+        else
+          echo ".env file exists, skipping...";
+        fi;
```

Comment:
This probably will need to move to configmap.

If a user needs to change URL or username etc. won't work.

Not sure how pytrakt works.
Is it autogenerated?

Reply:
pytrakt needs to generated locally using another system then modified it after the chart is installed. that is what upstream recommended for docker installation.

Reply:
quick question does the InitContainers run every time every chart startup or no? we can probably just disclose this as a one time initial setup process or just remove the env section and have users just modify both `.env` and `.pytrakt .json` after the chart gets installed, i documented this fact in the docs section.